woocommerce/woocommerce
# Create HPOS tables if they do not exist when enabling compatibility mode

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

If HPOS tables do not exist yet and we run the WP CLI command `wp wc hpos compatibility-mode enable`, it gives the following error-

> Error: HPOS tables do not exist.

- Creating the HPOS tables first, before enabling the compact mode if tables do not exist when `enable` command is run.
- No changes are made for the `disable` command.

Closes #51940 

### How to test the changes in this Pull Request:
- Delete the HPOS tables from your DB.
- Run the WP CLI command `wp wc hpos compatibility-mode enable` in `trunk` and you should see the error message.
- Run the WP CLI command `wp wc hpos compatibility-mode enable` in this branch.
    - You should not see any error message.
    - HPOS tables should be created in your DB.
    - Compatibility mode should be enabled.
